patches/minecraft/net/minecraft/tileentity/TileEntityBeacon.java.patch

```diff
@@ -9,6 +9,20 @@
                      {
                          this.field_146015_k = false;
                          this.field_174909_f.clear();
+                         break;
+                     }
+-
++                    float[] color = iblockstate.func_177230_c().getBeaconColor(iblockstate, field_145850_b, blockpos$mutableblockpos); if(color != null){afloat=color;}else{
+                     tileentitybeacon$beamsegment.func_177262_a();
+                     continue;
+                 }
+-
+-                afloat = EntitySheep.func_175513_a((EnumDyeColor)iblockstate.func_177229_b(BlockStainedGlassPane.field_176245_a));
++                }else
++                    afloat = EntitySheep.func_175513_a((EnumDyeColor)iblockstate.func_177229_b(BlockStainedGlassPane.field_176245_a));
```

Comment:
just add the `}else` directly on a new line on top of the old line.

that way this part is just +1 instead of -2 +2